usaspending_api/etl/management/commands/load_submission.py

```diff
@@ -564,6 +564,74 @@ def load_file_b(submission_attributes, prg_act_obj_cls_data, db_cursor):
     logger.info('Skipped a total of {} TAS rows for File B'.format(total_tas_skipped))
 
 
+def get_or_create_summary_award(awarding_agency=None, piid=None, fain=None,
+                                uri=None, parent_award_id=None, save=True):
+    """
+    Given a set of award identifiers and awarding agency information,
+    find a corresponding Award record. If we can't find one, create it.
+
+    Returns:
+        created: a list of new awards created, used to enable bulk insert
+        summary_award: the summary award that the calling process can map to
+    """
+    # If an award transaction's ID is a piid, it's contract data
+    # If the ID is fain or a uri, it's financial assistance. If the award transaction
+    # has both a fain and a uri, include both.
+    try:
+        lookup_kwargs = {"awarding_agency": awarding_agency}
+        for i in [(piid, "piid"), (fain, "fain"), (uri, "uri")]:
+            lookup_kwargs[i[1]] = i[0]
+            if parent_award_id:
+                # parent_award__piid
+                lookup_kwargs["parent_award_piid"] = parent_award_id
+
+        # Look for an existing award record
+        summary_award = Award.objects \
+            .filter(Q(**lookup_kwargs)) \
+            .filter(awarding_agency=awarding_agency) \
```

Comment:
The awarding agency shouldnt be part of the find (for finding a file C row from a file D)

But it should be used if there is a create.

Reply:
lines to remove
591
593-611

Reply:
Keeping the old filters to ensure we avoid creating duplicate awards when the row in question gets reloaded and still doesn't find a matching award that came in from the authoritative source